File: src/main/java/de/daniel/bactromod/impl/ModMenuIntegration.java
package de.daniel.bactromod.impl;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import de.daniel.bactromod.config.ConfigScreen;

public final class ModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return ConfigScreen::getConfigScreen;
    }
}

File: src/main/kotlin/de/daniel/bactromod/impl/ModMenuIntegration.kt
package de.daniel.bactromod.impl

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import de.daniel.bactromod.config.ConfigScreen

class ModMenuIntegration : ModMenuApi {

    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> = ConfigScreenFactory { ConfigScreen(it) }

}